File: day02/day02.py
#!/usr/bin/env python3
from collections import Counter
from itertools import combinations

data = open('input.txt').readlines()
twos = sum(1 for x in [set(Counter(y).values()) for y in data] if 2 in x)
threes = sum(1 for x in [set(Counter(y).values()) for y in data] if 3 in x)
print(twos * threes)
for x, y in combinations(data, 2):
    diff = [i for i in range(len(x)) if x[i] != y[i]]
    if len(diff) == 1:
        print(x[:diff[0]] + x[diff[0]+1:])
        break

File: day02/day02.rb
data = File.readlines('input.txt').map(&:chars)
counter = data.map{ |x| x.group_by &:itself}
twos = counter.count{ |x| x.any?{ |x,y| y.count == 2 }}
threes = counter.count{ |x| x.any?{ |x,y| y.count == 3 } }
p twos * threes
data.map{ |x| data.map { |y|
            a = x.zip(y).count{ |x,y| x!=y }
            (p x.join, y.join
             1/0) if a == 1
          }}

